.github/workflows/build-mixxx.yml

```diff
@@ -0,0 +1,110 @@
+          ${{ env.cache-name }}-${{ runner.os }}-${{ github.head_ref }}
+          ${{ env.cache-name }}-${{ runner.os }}
+    - name: macOS build environment cache
+      uses: actions/cache@v2
+      env:
+        cache-name: macOS build environment
+      with:
+        path: /Users/runner/buildenv
+        key: macOS-build-environment-${{ env.MIXXX_ENVNAME }}
+    - name: import code signing identity
+      env:
+        MACOS_CODESIGN_OPENSSL_PASSWORD: ${{ secrets.MACOS_CODESIGN_OPENSSL_PASSWORD }}
+        MACOS_CODESIGN_CERTIFICATE_PASSWORD: ${{ secrets.MACOS_CODESIGN_CERTIFICATE_PASSWORD }}
+      run: |
+        if [ -n "$MACOS_CODESIGN_OPENSSL_PASSWORD" ] && [ -n "$MACOS_CODESIGN_CERTIFICATE_PASSWORD" ]; then
+          # Decrypt the certificate
+          openssl enc -aes-256-cbc -d -md sha512 \
+            -k ${MACOS_CODESIGN_OPENSSL_PASSWORD} \
+            -in /Users/runner/work/mixxx/mixxx/cmake/macos_developer_id_codesign_certificate.p12.enc \
+            -out ~/certificate.p12
+
+          # Create a temporary keychain for the certificate and import it.
+          security create-keychain -p mixxx Mixxx.keychain
+          security unlock-keychain -p mixxx Mixxx.keychain
+          security import ~/certificate.p12 -k Mixxx.keychain \
+            -P ${MACOS_CODESIGN_CERTIFICATE_PASSWORD} -T /usr/bin/codesign
+          security set-key-partition-list -S apple-tool:,apple: -k mixxx Mixxx.keychain
+          # Add keychain to search list
+          security list-keychains -s Mixxx.keychain
+        fi
+    - name: build
+      run: |
+        mkdir -p /Users/runner/buildenv
+        export MIXXX_ENVPATH="/Users/runner/buildenv/$MIXXX_ENVNAME"
+        if [ ! -d "$MIXXX_ENVPATH" ]; then
+          # TODO: download from downloads.mixxx.org
+          curl "http://beasintheverb.me/${MIXXX_ENVNAME}.tar.gz" -o /Users/runner/buildenv/${MIXXX_ENVNAME}.tar.gz
+          tar xf /Users/runner/buildenv/${MIXXX_ENVNAME}.tar.gz -C /Users/runner/buildenv
+          rm /Users/runner/buildenv/${MIXXX_ENVNAME}.tar.gz
+        fi
+        export QT_DIR="$(find "${MIXXX_ENVPATH}" -type d -path "*/cmake/Qt5")"
+        export QT_QPA_PLATFORM_PLUGIN_PATH="$(find "${MIXXX_ENVPATH}" -type d -path "*/plugins")"
+
+        export PATH="${MIXXX_ENVPATH}/bin:$PATH"
+        ccache -M 5G
+
+        if [ -n "$MACOS_CODESIGN_OPENSSL_PASSWORD" ] && [ -n "$MACOS_CODESIGN_CERTIFICATE_PASSWORD" ]; then
+          export CMAKE_FLAGS_EXTRA="$CMAKE_FLAGS_EXTRA -DAPPLE_CODESIGN_IDENTITY=${APPLE_CODESIGN_IDENTITY}"
+        fi
+
+        cd /Users/runner/work/mixxx/mixxx
+        mkdir cmake_build
+        cd cmake_build
+        cmake -L $CMAKEFLAGS $CMAKEFLAGS_EXTRA -DCMAKE_PREFIX_PATH=${MIXXX_ENVPATH} -DQt5_DIR=${QT_DIR} ..
+        cmake --build . --parallel $(sysctl -n hw.physicalcpu)
+        cmake --build . --target benchmark
+        #TODO: fix disabled test
+        ctest --timeout 45 --exclude-regex DirectoryDAOTest.relocateDirectory
```

Comment:
Split this into a separate test step? Makes it easier to see what exactly failed.